File: ruby/test/wire/agent_imports_test.rb
# frozen_string_literal: true

require_relative "wiremock_test_case"

class AgentImportsWireTest < WireMockTestCase
  def setup
    super

    @client = Sikaru::Client.new(
      api_key: "<token>",
      base_url: WIREMOCK_BASE_URL
    )
  end

  def test_agent_imports_list_agent_imports_with_wiremock
    test_id = "agent_imports.list_agent_imports.0"

    @client.agent_imports.list_agent_imports(
      project_id: "project_id",
      request_options: {
        additional_headers: {
          "X-Test-Id" => "agent_imports.list_agent_imports.0"
        }
      }
    )

    verify_request_count(
      test_id: test_id,
      method: "GET",
      url_path: "/v1/projects/project_id/agent-imports",
      query_params: nil,
      expected: 1
    )
  end

  def test_agent_imports_create_agent_import_with_wiremock
    test_id = "agent_imports.create_agent_import.0"

    @client.agent_imports.create_agent_import(
      project_id: "project_id",
      improve: {
        mode: "mode",
        objective: "objective",
        promotion_gate: "promotionGate"
      },
      name: "name",
      request_options: {
        additional_headers: {
          "X-Test-Id" => "agent_imports.create_agent_import.0"
        }
      }
    )

    verify_request_count(
      test_id: test_id,
      method: "POST",
      url_path: "/v1/projects/project_id/agent-imports",
      query_params: nil,
      expected: 1
    )
  end
end

File: rust/tests/agent_imports_test.rs
use sikaru::prelude::*;

mod wire_test_utils;

#[tokio::test]
#[allow(unused_variables, unreachable_code)]
async fn test_agent_imports_list_agent_imports_with_wiremock() {
    wire_test_utils::reset_wiremock_requests().await.unwrap();
    let wiremock_base_url = wire_test_utils::get_wiremock_base_url();

    let mut config = ClientConfig {
        token: Some("<token>".to_string()),
        ..Default::default()
    };
    config.base_url = wiremock_base_url.to_string();
    let client = Sikaru::new(config).expect("Failed to build client");

    let result = client
        .agent_imports
        .list_agent_imports(&"project_id".to_string(), None)
        .await;

    assert!(result.is_ok(), "Client method call should succeed");

    wire_test_utils::verify_request_count("GET", "/v1/projects/project_id/agent-imports", None, 1)
        .await
        .unwrap();
}

#[tokio::test]
#[allow(unused_variables, unreachable_code)]
async fn test_agent_imports_create_agent_import_with_wiremock() {
    wire_test_utils::reset_wiremock_requests().await.unwrap();
    let wiremock_base_url = wire_test_utils::get_wiremock_base_url();

    let mut config = ClientConfig {
        token: Some("<token>".to_string()),
        ..Default::default()
    };
    config.base_url = wiremock_base_url.to_string();
    let client = Sikaru::new(config).expect("Failed to build client");

    let result = client
        .agent_imports
        .create_agent_import(
            &"project_id".to_string(),
            &CreateAgentImportRequest {
                improve: AgentImportImproveRequest {
                    mode: "mode".to_string(),
                    objective: "objective".to_string(),
                    promotion_gate: "promotionGate".to_string(),
                    ..Default::default()
                },
                name: "name".to_string(),
                eval_suites: None,
                model_capture: None,
                runner: None,
                source_refs: None,
            },
            None,
        )
        .await;

    assert!(result.is_ok(), "Client method call should succeed");

    wire_test_utils::verify_request_count("POST", "/v1/projects/project_id/agent-imports", None, 1)
        .await
        .unwrap();
}
